qiskit/circuit/singleton.py

```diff
@@ -0,0 +1,250 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Definition of the base singleton apparatus."""
+
+import operator
+
+from .instruction import Instruction
+from .gate import Gate
+
+# Summary
```

Comment:
Yeah, it was a comment just because I assumed it was for Qiskit library-author consumption and not public consumption, but I can stick it in the docstring properly.

Reply:
Done in 145decf53e.